src/stores/GuildMemberRoleStore.js

```diff
@@ -137,7 +148,13 @@ class GuildMemberRoleStore extends Collection {
    *   .then(member => console.log(`Member roles is now of ${member.roles.size} size`))
    *   .catch(console.error);
    */
-  set(roles, reason) {
+  set(roles = [], reason) {
+    for (let [Snowflake, Role] of this) {
+      if (Role.managed) {
+        if (typeof roles[0] === 'string' && !roles.includes(Snowflake)) roles.push(Snowflake);
+        else if (!roles.some(role => role.id === Snowflake)) roles.push(Snowflake);
```

Comment:
This will break when setting a collection. You might have to move this logic elsewhere.